### PR TITLE
Cleanups 1: Comments and formatting

### PR DESCRIPTION
**Pull request by StephanTLavavej**
* Detach comments.
  + This isn't attempting to be a global cleanup. Many of these will prevent the addition of `_EXPORT_STD` from causing wrapping, some fix wrapping that's already bad, some are for local consistency.
* `<string>`: Remove unnecessary comments for `to_string()`, `to_wstring()`, and `stoi()` etc. - they're just repeating the function names.
  + A few (like `stold()`) would also cause wrapping with `_EXPORT_STD`.
* `<cstddef>`: Remove SHOUTY RELICS and unnecessary comments, detach remaining comments.
* `<type_traits>`: Comment `_Refwrap_ctor_fun` as `// not defined`.
* `<random>`: Remove extra newline.
* `<random>`: Unwrap string literals for assertions.
* `<xutility>`: Shrink `clang-format off` region, add newline after declaring `ranges::less`.

### CODE REVIEW

File: stl/inc/random [hunk 1]
```diff
@@ -4218,10 +4218,8 @@ public:
         }
 
         void _Init(_Ty _Kx0, double _Px0) { // initialize
-            _STL_ASSERT(0.0 < _Kx0, "invalid max argument for "
-                                    "negative_binomial_distribution");
-            _STL_ASSERT(0.0 < _Px0 && _Px0 <= 1.0, "invalid probability argument for "
-                                                   "negative_binomial_distribution");
+            _STL_ASSERT(0.0 < _Kx0, "invalid max argument for negative_binomial_distribution");
```
Comment (strega-nil-ms) [inline stl/inc/random:4221]: I don't understand what changed here such that we can now fit this string on one line... shouldn't clang-format do a consistent thing here?

**Review by strega-nil-ms — APPROVED** [review]

File: stl/inc/random (hunk 1 above)
Comment (CaseyCarter) [reply stl/inc/random:4221]: clang-format will split-and-wrap overlong comments and string literals, but it will not merge comments and string literals on consecutive lines even if they wouldn't produce an overlong line. The idea begin that if you introduce line breaks in comments and string literals you do so to achieve some specific format that the tool shouldn't upset.

TLDR: Nothing changed here to allow the merge.

**Review by CaseyCarter — APPROVED** [review]

File: stl/inc/random (hunk 1 above)
Comment (strega-nil-ms) [reply stl/inc/random:4221]: Thanks for the explanation :)

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Review by raghavthind2005 — APPROVED** [review]
